packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,437 @@
+      isPillDeleted.current = true;
+    }
+  };
+
+  const handlePillListKeyDown = (event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const rollUpPill = filterPillsRef.current.querySelector('[data-terra-rollup-pill]');
+    setIsPillRemoved(false);
+    setShowRollupPillInteraction(false);
+
+    switch (event.keyCode) {
+      case KEY_RIGHT:
+        event.preventDefault();
+        focusNextNode(pills, rollUpPill, false);
+        break;
+      case KEY_LEFT:
+        event.preventDefault();
+        focusPreviousNode(pills, rollUpPill);
+        break;
+      case KEY_BACK_SPACE || KEY_DELETE:
+        event.preventDefault();
+        focusNodeAfterDelete(pills, rollUpPill);
+        break;
+      case KEY_HOME:
+        event.preventDefault();
+        setTabIndex('-1');
+        focusNode.current = 0;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+        focusCurrentNode();
+        break;
+      case KEY_END:
+        event.preventDefault();
+        setTabIndex('-1');
+        focusNode.current = pills.length - 1;
+        currentPill.current = pills[focusNode.current].id;
+        setTabIndex('0');
+        focusCurrentNode();
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleOnRemove = (pillKey, metaData, event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const targetId = event.target.parentElement.getAttribute('id');
+    const currentIndex = pills.findIndex((element) => element.id === targetId);
+    if (event.type === 'click') {
+      if (pills.length > 1) {
+        setTabIndex('-1');
+        if (currentIndex === 0) {
+          focusNode.current = 0;
+        } else {
+          focusNode.current = currentIndex - 1;
+        }
+        setTabIndex('0');
+      }
+      setIsPillRemoved(true);
+    }
+    focusPillsContainer();
+
+    if (onRemove) {
+      onRemove(pillKey, metaData);
+    }
+  };
+
+  // set the focus to current pill if the pill is clicked with mouse
+  const handleOnPillSelect = (pillRef) => {
+    if (pillRef.current && pillRef.current.children.length > 0) {
+      removedLabel.current = pillRef.current.children[0].innerText;
+    }
+  };
+
+  const handleOnSelectRollUp = (event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    if (isCollapsible && isCollapsed) {
+      if (event.type === 'keydown') {
+        isRollUpRemoved.current = true;
+        focusNode.current = pills.length;
+      } else {
+        setTabIndex('-1');
+        focusNode.current = pills.length;
+      }
+    } else {
+      setShowRollupPillInteraction(false);
+    }
+    setIsCollapsed(!isCollapsed);
+  };
+
+  const handlePillListOnblur = () => setContainerTabindex('-1');
+
+  const filterPillsProps = {};
+  filterPillsProps.onKeyDown = handlePillListKeyDown;
+  filterPillsProps.onBlur = handlePillListOnblur;
```

Comment:
Can we just set these directly when returning the component below on line 405?

Reply:
If we move the events directly inside the 'div' the linter throws error for  'non-interactive elements interaction'. This seemed a decent workaround instead of introducing disabling rules. Thoughts?

Reply:
Thanks @lokesh-0813 for the explanations as I was unaware of the possible linter error. I think it is better to keep it this way than to disable the rule.